configure.ac

```diff
@@ -691,6 +690,33 @@ if test "$ac_cv_func_getaddrinfo" = yes; then
   AC_DEFINE(HAVE_GETADDRINFO,1,[Define if you have the getaddrinfo function])
 fi
 
+AC_CACHE_CHECK([for copy_file_range], ac_cv_copy_file_range,
+[AC_RUN_IFELSE([AC_LANG_SOURCE([[
+#if defined(__linux__)
+#if !defined(_GNU_SOURCE)
+#define _GNU_SOURCE
+#endif
+#include <linux/version.h>
+#include <unistd.h>
+
+int main(void) {
+(void)copy_file_range(-1, 0, -1, 0, 0, 0);
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5,3,0)
+#error kernel too old;
```

Comment:
think it is usually just wrapped in `"` and no `;` like in https://gcc.gnu.org/onlinedocs/cpp/Diagnostics.html